spec/index.adoc

```diff
@@ -1675,27 +1675,32 @@ It is important to note the following for the usage of the `POST` method:
+The *size* of an HTTP request or response is the total number of bytes representing the names and values of HTTP headers and the HTTP body. Each request can specify a maximal size for the response from the remote HTTP server. The upper limit on the maximal size for the response is `2MB` (`2,000,000B`) and this value also applies if no maximal size value is specified.
 An error will be returned when the response is larger than the maximal size.
-The `2MiB` size limit also applies to the value returned by the `transform` function.
 
 The following parameters should be supplied for the call:
 
-- `url` - the requested URL. The URL may specify a custom port number.
-- `max_response_bytes` - optional, specifies the maximal size of the response in bytes. Any value less than or equal to `2MiB` is accepted. The call will be charged based on this parameter. If not provided, the maximum of `2MiB` will be used.
+- `url` - the requested URL. The URL must be valid according to https://www.ietf.org/rfc/rfc3986.txt[RFC-3986] and its length must not exceed 65534. The URL may specify a custom port number.
+- `max_response_bytes` - optional, specifies the maximal size of the response in bytes. If provided, the value must not exceed `2MB` (`2,000,000B`). The call will be charged based on this parameter. If not provided, the maximum of `2MB` will be used.
 - `method` - currently, only GET, HEAD, and POST are supported
 - `headers` - list of HTTP request headers and their corresponding values
 - `body` - optional, the content of the request's body
 - `transform` - an optional record that includes a function that transforms raw responses to sanitized responses, and a byte-encoded context that is provided to the function upon invocation, along with the response to be sanitized. If provided, the calling canister itself must export this function.
 
+Cycles to pay for the call must be explicitly transferred with the call, i.e., they are not deducted from the caller's balance implicitly (e.g., as for inter-canister calls).
+
 The returned response (and the response provided to the `transform` function, if specified) contains the following fields:
 
 - `status` - the response status (e.g., 200, 404)
 - `headers` - list of HTTP response headers and their corresponding values
 - `body` - the response's body
 
-The `transform` function may, for example, transform the body in any way, add or remove headers, modify headers, etc.
-When the transform function was invoked due to a canister HTTP request, the caller's identity is the principal of the management canister.
+The `transform` function may, for example, transform the body in any way, add or remove headers, modify headers, etc. The maximal size of the blob produced by the `transform` function is `2MB` (`2,000,000B`). When the transform function was invoked due to a canister HTTP request, the caller's identity is the principal of the management canister.
```

Comment:
Also

> When the transform function was invoked due to a canister HTTP request, the caller's identity is the principal of the management canister.
This is mentioned below in a separate paragraph. So I'd either remove it from here or remove the other paragraph (I prefer keeping the separate paragraph as it is more informative, or merging the two).

Reply:
I changed "The maximal size of the blob" to "The maximal number of bytes representing the response". I agree that talking about a blob is confusing given `ic.did`. On the other hand, the slack we have above `2MB` might not be enough for the transform function, e.g., because a custom candid encoder in the transform might include a huge type table (if no transform function is provided, then the system uses a candid encoder for which the slack is sufficient).

I removed the note on caller's identity in transform because the sentence "The identity of the caller for a valid invocation of the `transform` function is `aaaaa-aa`." is wrong. The caller is `aaaaa-aa` only if the transform function is invoked by the system for canister http_request calls, but a valid invocation need not be performed by the system only.

I added the (useful) sentence "This information can be used by developers to implement access control mechanism for this function." from the note into the main text.

Reply:
Thanks! Just added a small writing fix but otherwise it looks good to me.